src/libraries/Common/src/System/Net/Http/X509ResourceClient.cs

```diff
@@ -80,6 +81,73 @@ internal static partial class X509ResourceClient
             return null;
         }
 
+        private const string SocketsHttpHandlerTypeName = "System.Net.Http.SocketsHttpHandler, System.Net.Http";
```

Comment:
Wondering what was the reason for this design where the client has to know the right type, assembly syntax and you end up with these long string instead of having attribute constructor with multiple parameters.

Reply:
The design was to use strings of the exact same format that you'd pass to the `Type.GetType(string)` method.

These are primarily intended to be used in generated code (the usages in the BCL are due to our own layering problems mainly) and requiring the assembly-qualified name format makes it much easier for the runtimes to find the types to insert into the signatures.